Description: Path of utsrelease.h changed to
 $(KERNEL_DIR)/generated/utsrelease.h in linux kernel >= 2.6.33
 Credits to Harald Jenny <harald@a-little-linux-box.at> for the heads up
Forwarded: yes
Author: أحمد المحمودي (Ahmed El-Mahmoudy) <aelmahmoudy@sabily.org>
--- a/drivers/Makefile
+++ b/drivers/Makefile
@@ -82,7 +82,7 @@
 	    $(RM) ${DESTDIR}/dev/slusb$(minor) ; ) echo -n
 
 else
-ifeq ($(shell grep 'UTS.*"2\.6' $(KERNEL_DIR)/include/linux/utsrelease.h $(KERNEL_DIR)/include/linux/version.h 2>/dev/null),)
+ifeq ($(shell grep 'UTS.*"2\.6' $(KERNEL_DIR)/include/generated/utsrelease.h $(KERNEL_DIR)/include/linux/utsrelease.h $(KERNEL_DIR)/include/linux/version.h 2>/dev/null),)
 
 slusb-objs:= old_st7554.o
 obj:=.
--- a/drivers/kernel-ver.c
+++ b/drivers/kernel-ver.c
@@ -5,9 +5,13 @@
 
 #include <stdio.h>
 #include <linux/version.h>
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,33)
+#include <generated/utsrelease.h>
+#else
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,18)
 #include <linux/utsrelease.h>
-#endif
+#endif // KERNEL_VERSION(2,6,18)
+#endif // KERNEL_VERSION(2,6,33)
 
 int main()
 {
